crates/admin/src/rest_api/invocations.rs

```diff
@@ -8,121 +8,23 @@
 // the Business Source License, use of this software will be governed
 // by the Apache License, Version 2.0.
 
-use super::error::*;
-use crate::generate_meta_api_error;
-use crate::rest_api::create_envelope_header;
-use crate::state::AdminServiceState;
 use axum::Json;
 use axum::extract::{Path, Query, State};
 use axum::http::StatusCode;
 use okapi_operation::*;
+use serde::Deserialize;
+
 use restate_admin_rest_model::invocations::RestartAsNewInvocationResponse;
-use restate_types::identifiers::{
-    DeploymentId, InvocationId, PartitionProcessorRpcRequestId, WithPartitionKey,
-};
+use restate_types::identifiers::{DeploymentId, InvocationId, PartitionProcessorRpcRequestId};
 use restate_types::invocation::client::{
     self, CancelInvocationResponse, InvocationClient, KillInvocationResponse,
     PauseInvocationResponse, PurgeInvocationResponse, ResumeInvocationResponse,
 };
-use restate_types::invocation::{InvocationTermination, PurgeInvocationRequest, TerminationFlavor};
 use restate_types::journal_v2::EntryIndex;
-use restate_wal_protocol::{Command, Envelope};
-use serde::Deserialize;
-use std::sync::Arc;
-use tracing::warn;
 
-#[derive(Debug, Default, Deserialize, JsonSchema)]
-pub enum DeletionMode {
-    #[default]
-    #[serde(alias = "cancel")]
-    Cancel,
-    #[serde(alias = "kill")]
-    Kill,
-    #[serde(alias = "purge")]
-    Purge,
-}
-#[derive(Debug, Default, Deserialize, JsonSchema)]
-pub struct DeleteInvocationParams {
-    pub mode: Option<DeletionMode>,
-}
-
-/// Terminate an invocation
-#[openapi(
-    summary = "Delete an invocation",
-    deprecated = true,
```

Comment:
This API was deprecated in 1.4.0. If we remove it with 1.6.0, then we certainly need to add a release note to make people aware. Additionally, we need to check https://github.com/restatedev/restate/blob/b9b8f281c4f0133284a2c60350c510443162a7d5/crates/admin-rest-model/src/version.rs#L29 and https://github.com/restatedev/restate/blob/b9b8f281c4f0133284a2c60350c510443162a7d5/crates/admin/src/rest_api/version.rs#L21 whether we need to remove support for AdminApiVersion::V2 because we removed the old delete invocation endpoint. We should then also update https://github.com/restatedev/restate/blob/b9b8f281c4f0133284a2c60350c510443162a7d5/cli/src/clients/admin_client.rs#L34 accordingly if a bump is needed.

Reply:
I actually discussed this with him before I remove it. I was confused because the was 2 endpoints to delete an invocation. So he confirmed that it's no longer used (same for the UI).

Regarding the version, thank you for pointing this out. I think we should indeed set the MIN_ADMIN_API_VERSION to V3